eth/handler_eth.go

```diff
@@ -180,6 +180,12 @@ func (h *ethHandler) handleBodies(peer *eth.Peer, txs [][]*types.Transaction, un
 // handleBlockAnnounces is invoked from a peer's message handler when it transmits a
 // batch of block announcements for the local node to process.
 func (h *ethHandler) handleBlockAnnounces(peer *eth.Peer, hashes []common.Hash, numbers []uint64) error {
+	// Drop all incoming block announces from the p2p network if
+	// the chain already entered the pos stage and disconnect the
+	// remote peer.
+	if h.merger.PoSFinalized() {
+		return errors.New("unexpected block announces")
+	}
```

Comment:
This is an interesting question whether it should be done or not. When I start up a new peer, it won't know if it's in PoS or not yet, so it might do something funky nuking it off the network. Shouldn't we perhaps just ignore announces instead of returning an error and let an eventual protocol update remove this feature? Seems a bit safer vs. an unexpected situation leading to disconnects.